regolith/validators.py

```diff
@@ -109,4 +113,49 @@ def ensure_email(email):
     'databases': (always_false, ensure_databases),
     'stores': (always_false, ensure_stores),
     'email': (always_false, ensure_email),
-    }
+}
+
+
+# TODO: run this as a lint like thing rather than raising at the first error
+def validate_schema(record, schema, keys=()):
+    """Validate a database record against a schema
+
+    Parameters
+    ----------
+    record : dict
+        The database record to be tracked
+    schema : dict
+        The schema to validate the record against
+    keys : tuple
+        The sequential keys for the data being accessed, used for
+        printing/debuging errors
+    """
+    if isinstance(record, dict):
+        total_keys = set(schema.keys())
+        remove_keys = ['required', 'type', 'description']
+        for k in remove_keys:
+            if k in total_keys:
+                total_keys.remove(k)
+        total_keys.update(set(record.keys()))
+
+        for k in total_keys:
+            if k not in schema:
+                pass
+            if k not in record and schema[k].get('required', False):
+                raise ValueError('{} is required in {}'.format(k, keys))
+            elif k in record and k in schema:
+                validate_schema(record[k], schema[k], keys + (k, ))
+    elif isinstance(record, collections.Iterable) and not isinstance(record,
```

Comment:
Does this really include set and generators? Or is this supposed to be `collections.abc.Sequence`?